Actual Main Branch

Fooot-Code opened the pull request:
All of the code associated with april tags as well as some ai models for coral and algae to use with photon vision. Mainly being merged because of outdated main branch.

Fooot-Code committed:
Separated files

ThePrinterFRC1799 committed:
1 line in 40 mins

ThePrinterFRC1799 committed:
D1 Crashout

Fooot-Code committed:
Publishes Correctly

Able to publish data from pose estimator

Fooot-Code committed:
Added Note Camera Class

Fooot-Code committed:
Corrected link

Fooot-Code committed:
oh my god

Fooot-Code committed:
quicksave b4 restart

Fooot-Code committed:
just you kn

Fooot-Code committed:
Before Restart Again

Fooot-Code committed:
Install stuff

Fooot-Code committed:
Works well, just make sure to exit with "q"

Fooot-Code committed:
Just Figured out my code is useless cuh

Fooot-Code committed:
Arducam Testing

Fooot-Code committed:
documentation

Fooot-Code committed:
Best Coder on the Team

Fooot-Code committed:
WE HAVE TO USE THIS

SORRY MILO BUT THIS SHIII GASS CREDIT TO 6647 THEY THE GOATSSSSSSS

Fooot-Code committed:
quicksave

Fooot-Code committed:
good remnea

Fooot-Code committed:
added __rect_inside_another_rect

Fooot-Code committed:
Added some theoretical working stuff

Fooot-Code committed:
More theoretical Stuff

Fooot-Code committed:
Full class functions written

Fooot-Code committed:
testing well 2 cams work

Fooot-Code committed:
Implementation Complete

Fooot-Code committed:
Cont Delete

Fooot-Code committed:
added model

Fooot-Code committed:
more fun :?

Fooot-Code committed:
ray

twisterjafla committed:
needs to be checked but should work

twisterjafla committed:
dont use, will need to be rewritten with wpi classes. math is way worse than i thought

twisterjafla committed:
should work now

Fooot-Code committed:
implemented hitboxes

Fooot-Code committed:
Merge pull request #14 from FRC-1799/mathForBrennon

Hitboxes and Math

Fooot-Code committed:
Past reef grab

Fooot-Code committed:
Changed functs even  tho yo no necesito

Fooot-Code committed:
brain ouch

Fooot-Code committed:
deletes the libs so it wont cause other computers conflicts

Fooot-Code committed:
Refactoring

Fooot-Code committed:
CHANGED VALUES FOR REEF LOCATIONS HERE

Fooot-Code committed:
Hitboxes are correct enough

Fooot-Code committed:
Deleted Previous Coral Camera

Fooot-Code committed:
Algea stuff

Fooot-Code committed:
Working with hitboxes

ThePrinterFRC1799 committed:
hitbox stuff

Fooot-Code committed:
refactored and cool

twisterjafla committed:
brennon code fix

twisterjafla committed:
works

twisterjafla committed:
Merge branch 'Reef-Detection' into fixing-hitboxes

twisterjafla committed:
added algae

twisterjafla committed:
Merge branch 'fixing-hitboxes' of https://github.com/FRC-1799/Note-Detection into fixing-hitboxes

twisterjafla committed:
added mirroring

twisterjafla committed:
fixed mirroring

ThePrinterFRC1799 committed:
updated for april testing

ThePrinterFRC1799 committed:
Camera 0 Config

ThePrinterFRC1799 committed:
Renamed and Added A Config

ThePrinterFRC1799 committed:
Cam 2 Config

ThePrinterFRC1799 committed:
Configs into a folder

ThePrinterFRC1799 committed:
Pushed for testing

Fooot-Code committed:
Merge branch 'Reef-Detection' into fixing-hitboxes

Fooot-Code committed:
Merge pull request #15 from FRC-1799/fixing-hitboxes

Fixing hitboxes

ThePrinterFRC1799 committed:
wow

Fooot-Code committed:
quick

Fooot-Code committed:
Test Ready???

ThePrinterFRC1799 committed:
coding fun part 1

ThePrinterFRC1799 committed:
Forgot to pull before making changes

ThePrinterFRC1799 committed:
quick

Fooot-Code committed:
good for comp

ThePrinterFRC1799 committed:
for comp pt 2

Fooot-Code committed:
Multiprocessing

ThePrinterFRC1799 committed:
Comp Changes

ThePrinterFRC1799 committed:
Merge remote-tracking branch 'origin/No-Multiprocessing' into Reef-Detection

ThePrinterFRC1799 committed:
Yessir

ThePrinterFRC1799 committed:
better for testing

ThePrinterFRC1799 committed:
Coral debug implementation

Fooot-Code committed:
Testing on computer

Fooot-Code committed:
Test stuff to be refactored

ThePrinterFRC1799 committed:
Refactored for some tests

Fooot-Code committed:
Configs and Refactor Testing

Fooot-Code committed:
Merge branch 'Reef-Detection' into No-Multiprocessing

Fooot-Code committed:
Merge pull request #16 from FRC-1799/No-Multiprocessing

Improving Logic and Refactoring

Fooot-Code committed:
im dumb so here

Fooot-Code committed:
Merge branch 'Reef-Detection' of https://github.com/FRC-1799/Note-Detection into Reef-Detection

Fooot-Code committed:
Removed Multiprocessing

Fooot-Code committed:
Comments and Documentation

Fooot-Code committed:
Refactoring and some testing

Fooot-Code committed:
file transfer dont worry about it

ThePrinterFRC1799 committed:
Revert "file transfer dont worry about it"

This reverts commit 16a4be836b0b98499c985f17f3634cb8bd87f01e.

ThePrinterFRC1799 committed:
ahhhh

Fooot-Code committed:
refactored a little

Fooot-Code committed:
Robot publish  good and bad

ThePrinterFRC1799 committed:
Different tables and timestamp other location

Fooot-Code committed:
Publishig workes

ThePrinterFRC1799 committed:
Rotation changed

Fooot-Code committed:
Name changes

Fooot-Code committed:
Merge branch 'Reef-Detection' into Test-Refactoring

Fooot-Code committed:
Merge pull request #17 from FRC-1799/Test-Refactoring

Refactoring and Tested

ThePrinterFRC1799 committed:
fixed indentation

ThePrinterFRC1799 committed:
uncommented

ThePrinterFRC1799 committed:
PyUDev implementation

ThePrinterFRC1799 committed:
better code

ThePrinterFRC1799 committed:
me too

Fooot-Code committed:
Name change

Fooot-Code committed:
Reverted name change

Fooot-Code committed:
Tested and Works :>

ThePrinterFRC1799 committed:
test implementation

Fooot-Code committed:
Ray works

Fooot-Code committed:
marks algae and coral correctly

ThePrinterFRC1799 committed:
Test Ready probably

ThePrinterFRC1799 committed:
umm

ThePrinterFRC1799 committed:
ahh test divide remove later

Fooot-Code committed:
test stuff

twisterjafla committed:
added start of rule

Fooot-Code committed:
Merge branch 'Coral-Detection-Theoretical-Test' of https://github.com/FRC-1799/Note-Detection into Coral-Detection-Theoretical-Test

Fooot-Code committed:
quicksand

Fooot-Code committed:
quicksave

ThePrinterFRC1799 committed:
removed i think

Fooot-Code committed:
updated nt table stuff

ThePrinterFRC1799 committed:
constants upgrade

Fooot-Code committed:
constants

Fooot-Code committed:
moved table and it works now :shrug:

Fooot-Code committed:
changed ip back

Fooot-Code committed:
TSPMOFRFROCNC

ThePrinterFRC1799 committed:
Converted to RKNN Format

twisterjafla commented on main.py:
A) Dont define functions inside other functions unless said funtions are lambdas. it is harder to read and has no benifits.
B) put a small sleep in the while true. since this code doesnt do much actual calculation it can loop realy fast but those loops are unnecessary. If you wanna be fancy you could make the update loop run at a set 50 hz but honestly a 10-15 ms sleep should do the job just fine. Anything above 50 hz is literally useless since the robot updates at that speed anyway. Having a wait is better because it keeps the code from doing unnecessary polling which in turn reduces the power consumption, processing consumption and keeps the pis cooler.

twisterjafla commented on Classes/AprilTagCamera.py:
```diff
@@ -0,0 +1,74 @@
+        self.cameraName = cameraName
+        self.camera = PhotonCamera(self.cameraName)
+        self.estimator = PhotonPoseEstimator(
+            apriltag.AprilTagFieldLayout.loadField(apriltag.AprilTagField.k2025ReefscapeWelded),
```
dont hardcode the field map. i would make it an argument too the camera so that this object can be reused in future years. You can even default it too 2025 if you realy want too but i wouldnt.

twisterjafla commented on 98-reef-rule.rules:
This file doesnt need to exist

twisterjafla reviewed (changes requested):
Looks good in general. i requested a few changes but nothing significant. mainly just formatting. Additionally if you wouldn't mind going through and removing all of the big comment blocks. A little bit of saved code here and there is fine but there is a lot of it in this repo and it is just a bit confusing.

Fooot-Code committed:
Implemented Requested Changes

Deleted large commented blocks, removed sleep function in while loop, moved function inside of a function outside of the function, and removed the rule file.

twisterjafla reviewed (changes requested):
the main loop still needs a sleep

Fooot-Code committed:
I definitely know how to read

Fooot-Code committed:
INDENT

twisterjafla reviewed (approved):
Looks good
